### ISSUE TITLE
java.lang.ArrayStoreException for jvmArgs or systemProperties

### ISSUE THREAD

**Issue by bzamecnik** [saladinkzn/GRefresher#1]
When running with `re-run` and `jvmArgs` or `systemProperties` are specified.
```
def configFile = "foo.conf"
jvmArgs = ["-Dconfig.file=${configFile}"]
// or
systemProperties = ["config.file": configFile]
```

I get this error:

```
Press 'q' or 'Q' to stop application or any other key to restart
Exception in thread "Thread-11" java.lang.ArrayStoreException
	at java.util.ArrayList.toArray(ArrayList.java:408)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1007)
	at java_lang_ProcessBuilder$start$1.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:117)
	at ru.shadam.grefresher.GRefresherTask$_startProcess_closure4.doCall(GRefresherTask.groovy:109)
	at ru.shadam.grefresher.GRefresherTask$_startProcess_closure4.doCall(GRefresherTask.groovy)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1019)
	at groovy.lang.Closure.call(Closure.java:426)
	at groovy.lang.Closure.call(Closure.java:420)
	at groovy.lang.Closure.run(Closure.java:507)
	at java.lang.Thread.run(Thread.java:745)
```

System info:
```

------------------------------------------------------------
Gradle 2.11
------------------------------------------------------------

Build time:   2016-02-08 07:59:16 UTC
Build number: none
Revision:     584db1c7c90bdd1de1d1c4c51271c665bfcba978

Groovy:       2.4.4
Ant:          Apache Ant(TM) version 1.9.3 compiled on December 23 2013
JVM:          1.8.0_77 (Oracle Corporation 25.77-b03)
OS:           Mac OS X 10.11.6 x86_64
```

**Comment by bzamecnik** [comment]
The problem was that "-Dconfig.file=${configFile}" is not java.lang.String but `org.codehaus.groovy.runtime.GStringImpl` and there's not implicit conversion.

**Comment by bzamecnik** [comment]
Still there's a problem when I pass strings to `systemProperties` they get joined at `GRefresherTask.groovy:109` into `GStringImpl` which does not get converted to String and then results in `ArrayStoreException`. This exception means that the list that's about to be converted to array is not uniformly typed.

The solution would be to convert all the values to strings.

**Comment by saladinkzn** [comment]
I published version 0.2 of GRefresher which addresses this issue too.
